#24 지역 조회시 redis에 ResponseEntity 저장하는것이 아닌 지역 클래스로 저장하도록 수정

0hun opened the pull request:
redis에 저장하는할때는 ResponseEntity가 아닌 해당 객체로 저장을 해야하므로 수정

0hun committed:
#24 지역 조회시 redis에 ResponseEntity 저장하는것이 아닌 지역 클래스로 저장하도록 수정

redis에 저장하는할때는 ResponseEntity가 아닌 해당 객체로 저장을 해야하므로 수정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/controller/RegionController.java:
```diff
@@ -23,27 +24,19 @@
    */
-
-  @Cacheable(value = "regions", key = "#name", cacheManager = "redisCacheManager")
+  @Cacheable(value = "regions.name", key = "#name", cacheManager = "redisCacheManager")
```
여기도 캐싱은 service에 해주세요~

0hun replied:
넵 알겠습니다.

0hun committed:
Merge branch 'issue/23' into issue/24

0hun committed:
#23 리뷰 반영

스프링에는 기본 ObjectMapper가 bean으로 등록되어있기 때문에 설정에서 삭제
서비스 레이어에서 캐시하도록 수정 및 만료시간 설정

0hun committed:
Merge branch 'issue/24' of https://github.com/f-lab-edu/hi-bus-go into issue/24

0hun committed:
#24 리뷰 반영

서비스 영역에서 캐시하기때문에 주석 위치 수정

f-lab-dev commented on src/main/java/com/younghun/hibusgo/service/RegionService.java:
```diff
@@ -18,7 +21,9 @@ public Region findById(int id) {
+  @Scheduled(fixedDelay = 300000) //5분마다 캐시에 저장
+  @Cacheable(value = "regions.name", key = "#name", cacheManager = "redisCacheManager")
+  public Optional<List<Region>> searchByName(String name) {
+    return Optional.ofNullable(regionMapper.searchByName(name));
```
이럴때는 리턴 타입이 리스트이기때문에 리턴값이 없다면 빈 리스트를 리턴해주는게 더 복잡도도 줄어들고 좋습니다~

0hun committed:
#24 캐시 설정

@Scheduled는 파라미터가 없는 메소드에만 적용가능하여, 공통 캐시 설정으로 수정
지역 정보는 변하지 않는 정보이기 때문에 공통 캐시 설정을 적용

0hun committed:
#24 리뷰 반영

리스트 값이 없다면 빈 리스트를 넘겨주도록 수정

0hun committed:
#24 리뷰 반영

ResponseEntity 리턴 타입 수정

0hun committed:
Merge branch 'issue/23' into issue/24

0hun committed:
#24 지역정보 없을시 http 404 return

데이터가 없을 경우 404 리턴 하도록 수정
